### ISSUE TITLE
Secure REST server error - The current identity has not been registered

### ISSUE THREAD

**Issue by jdockter** [hyperledger/composer#1784]
I have configured my local REST server to use Google auth and Mongo DB.  I have followed the deployment steps and can authenticate my Google ID putting the default Wallet into the Mongo DB.  I then create a participant and issue an identity for that participant.  I add the identity to the wallet and set that identity as the default.  I was able to do a system/ping to return the current participant.  However, once I started invoking transaction I soon hit an error 'The current identity has not been registered'.

## Context
```jpd@ubuntu:~/rest-server$composer -v
composer-cli                   v0.11.1
composer-admin                 v0.11.1
composer-client                v0.11.1
composer-common                v0.11.1
composer-runtime-hlf           v0.11.1
composer-runtime-hlfv1         v0.11.1
```

Environment Variables text file...
```COMPOSER_CONNECTION_PROFILE=hlfv1
COMPOSER_BUSINESS_NETWORK=customs-network
COMPOSER_ENROLLMENT_ID=admin
COMPOSER_ENROLLMENT_SECRET=adminpw
COMPOSER_NAMESPACES=never
COMPOSER_SECURITY=true
COMPOSER_CONFIG='{
  "connectionProfiles": {
    "hlfv1": {
      "name": "hlfv1",
      "description": "Hyperledger Fabric v1.0",
      "type": "hlfv1",
      "keyValStore": "/home/composer/.composer-credentials",
      "timeout": 300,
      "orderers": [
        {
          "url": "grpc://orderer.example.com:7050"
        }
      ],
      "channel": "composerchannel",
      "mspID": "Org1MSP",
      "ca": {
        "url": "http://ca.org1.example.com:7054",
        "name": "ca.org1.example.com"
      },
      "peers": [
        {
          "requestURL": "grpc://peer0.org1.example.com:7051",
          "eventURL": "grpc://peer0.org1.example.com:7053"
        }
      ]
    }
  }
}'
COMPOSER_DATASOURCES='{
  "db": {
    "name": "db",
    "connector": "mongodb",
    "host": "mongo"
  }
}'
COMPOSER_PROVIDERS='{
  "google": {
    "provider": "google",
    "module": "passport-google-oauth2",
    "clientID": "xxxxx",
    "clientSecret": "xxxxx",
    "authPath": "/auth/google",
    "callbackURL": "http://localhost:3000/auth/google/callback",
    "scope": "https://www.googleapis.com/auth/plus.login",
    "successRedirect": "http://localhost:3000/explorer",
    "failureRedirect": "/auth/google/failure"
  }
}'
```
```
jpd@ubuntu:~/rest-server$ composer network ping -n customs-network -p hlfv1 -i jpdslp -s psoIOWkHttJI

The connection to the network was successfully tested: customs-network
    version: 0.11.1
    participant: com.ibm.customs.SourceLogisticProvider#jdockter@us.ibm.com

Command succeeded
```

Transactions along with GET system/ping does not work, returns Error: The current identity has not been registered:jpdslp, even thought CLI and Mongo DB show it has been stored

```
_id:5988e602edd5f6001671ad11
enrollmentID:"jpdslp"
enrollmentSecret:"psoIOWkHttJI"
data:Object
walletId:5988e596edd5f6001671ad0f
```

## Expected Behavior
Secure REST server should keep my identity registered with the REST server.

## Actual Behavior
Seems to lose the registered identity after initial setup for secure REST server.

## Steps to Reproduce
1.  Can provide clientID and clientSecret if need to recreate exact settings
2.  Followed steps outlined in Composer docs here:
https://hyperledger.github.io/composer/integrating/enabling-rest-authentication.html
https://hyperledger.github.io/composer/integrating/deploying-the-rest-server.html

## Existing issues

- [x] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [x] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [x] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

Might be similar to issue https://github.com/hyperledger/composer/issues/1776

## Your Environment

* Version used: Composer v0.11.1
* Environment name and version (e.g. Chrome 39, node.js 5.4): Chrome
* Operating System and version (desktop or mobile): Ubuntu 16.0.4
* Link to your project: Can provide if needed.

**Comment by jdockter** [comment]
@mahoney1 please see the issue above and if you need any information to recreate I can provide it, thanks

**Comment by davidkel** [comment]
@jdockter This at first glance looks like its related to #1731. What concerns me is more is that you appear to be using a persistent storage facility and I would have expected the users certificates to have been stored there as well. Can you confirm that the certificates have been stored ? Although the original secret is saved and you can enroll it again, if that user had a limitation on the number of enrolments then it's important that certificates are persisted.

**Comment by davidkel** [comment]
Can you confirm that it was working then after a period of time it stopped working ? It looks like it does store the certificate information otherwise initial requests wouldn't work but then it seems like for some reason it decides to re-enroll the user. You can check that by looking in the ca logs to see the enrollments being done so if you can confirm that by pasting on the ca logs ?
`docker logs ca.org1.example.com 2>logs.txt` if you are using the development server.

**Comment by davidkel** [comment]
Enrollment 1
![image](https://user-images.githubusercontent.com/25582377/29074657-0682a5ec-7c48-11e7-8866-9caec74d2e4d.png)

**Comment by davidkel** [comment]
Unexpected enrollment 2
![image](https://user-images.githubusercontent.com/25582377/29074699-1f7151d4-7c48-11e7-8a28-44e16ac0d7a1.png)

**Comment by jdockter** [comment]
@davidkel running secure REST server without MongoDB seems to be working.  Per our Slack conversation it looks like with MongoDB the same identity is getting enrolled many times after a period of time with statements from the CA showing the following for identity TestMe:

```
	Line 4: {"id":"TestMe","type":"client","affiliation":"org1","max_enrollments":0,"attrs":[],"caName":"ca.org1.example.com"}
	Line 10: 2017/08/08 12:53:53 [DEBUG] Received registration request from admin: &{RegistrationRequest:{Name:TestMe Type:client Secret:<<user-specified>> MaxEnrollments:0 Affiliation:org1 Attributes:[] CAName:ca.org1.example.com}}
	Line 17: 2017/08/08 12:53:53 [DEBUG] Registering user id: TestMe
	Line 19: 2017/08/08 12:53:53 [DEBUG] DB: Getting identity TestMe
	Line 20: 2017/08/08 12:53:53 [DEBUG] DB: Add identity TestMe
	Line 21: 2017/08/08 12:53:53 [DEBUG] Successfully added identity TestMe to the database
	Line 29: 2017/08/08 12:56:49 [DEBUG] DB: Getting identity TestMe
	Line 30: 2017/08/08 12:56:49 [DEBUG] DB: Login user TestMe with max enrollments of -1 and state of 0
	Line 31: 2017/08/08 12:56:49 [DEBUG] Successfully incremented state for identity TestMe to 1
	Line 32: 2017/08/08 12:56:49 [DEBUG] DB: identity TestMe successfully logged in
	Line 43: 2017/08/08 12:56:49 [DEBUG] csrAuthCheck: enrollment ID=TestMe, CommonName=TestMe, Subject=<nil>
	Line 43: 2017/08/08 12:56:49 [DEBUG] csrAuthCheck: enrollment ID=TestMe, CommonName=TestMe, Subject=<nil>
	Line 55: 2017/08/08 13:31:52 [DEBUG] DB: Getting identity TestMe
	Line 56: 2017/08/08 13:31:52 [DEBUG] DB: Login user TestMe with max enrollments of -1 and state of 1
	Line 57: 2017/08/08 13:31:52 [DEBUG] Successfully incremented state for identity TestMe to 2
	Line 58: 2017/08/08 13:31:52 [DEBUG] DB: identity TestMe successfully logged in
	Line 69: 2017/08/08 13:31:52 [DEBUG] csrAuthCheck: enrollment ID=TestMe, CommonName=TestMe, Subject=<nil>
	Line 69: 2017/08/08 13:31:52 [DEBUG] csrAuthCheck: enrollment ID=TestMe, CommonName=TestMe, Subject=<nil>
```

Also in MongoDB extra private and public keys are added to TestMe.
![mongo_doc](https://user-images.githubusercontent.com/22593763/29076973-a11c1a9c-7c1c-11e7-9e93-cb8b85d41602.jpg)

**Comment by jdockter** [comment]
Seems like this is also tied to https://github.com/hyperledger/composer/issues/1731

**Comment by sstone1** [comment]
@jdockter should be fixed in v0.11.3, please can you give it a try?

**Comment by DennisM330** [comment]
I just tried this with v0.11.3. composer identity list shows my identity as registered and active. 
However, when I try the ping I get the following error. Using mongo also as the persistence.
{
  "error": {
    "statusCode": 500,
    "name": "Error",
    "message": "Error trying to ping. Error: Error trying to query chaincode. Error: chaincode error (status: 500, message: Error: The current identity has not been registered:drmiller)",
    "stack": "Error: Error trying to ping. Error: Error trying to query chaincode. Error: chaincode error (status: 500, message: Error: The current identity has not been registered:drmiller)\n    at _checkRuntimeVersions.then.catch (/home/student/.nvm/versions/node/v6.10.3/lib/node_modules/composer-rest-server/node_modules/composer-connector-hlfv1/lib/hlfconnection.js:696:34)"
  }
}

**Comment by DennisM330** [comment]
Appears my v0.11.3 upgrade to not update all the prereq packages for the rest server. I fixed my problem so this functionality for this issue was indeed addressed in v0.11.3
